adapters/improvedigital/improvedigital.go

```diff
@@ -180,33 +179,84 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server co
 	return bidder, nil
 }
 
-func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
-	for _, imp := range imps {
-		if imp.ID == impID {
-			if imp.Banner != nil {
-				return openrtb_ext.BidTypeBanner, nil
-			}
-
-			if imp.Video != nil {
-				return openrtb_ext.BidTypeVideo, nil
-			}
+func getBidType(bid openrtb2.Bid, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	// there must be a matching imp against bid.ImpID
+	imp, err := findImpByID(bid.ImpID, imps)
+	if err != nil {
+		return "", err
+	}
```

Comment:
```
func findImpByID(impID string, imps []openrtb2.Imp) (openrtb2.Imp, error) {
	for _, imp := range imps {
		if imp.ID == impID {
			return imp, nil
		}
	}
	return openrtb2.Imp{}, &errortypes.BadServerResponse{
		Message: fmt.Sprintf("Failed to find impression for ID: \"%s\"", impID),
	}
}
```

As per code setup, `imps` will be iterated every time to find corresponding matching `imp` for `bid.ImpID` 
Instead would recommend to create map[int]openrtb.Imp where map key will be imp id. This will reduce iteration needed to find `imp` for `bid.ImpID`

Reply:
Code updated to use map as suggested